packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/boolean.rs

```diff
@@ -58,11 +56,8 @@ pub struct BooleanState {
 impl BooleanState {
     fn new() -> Self {
         BooleanState {
-            value: GeneralBooleanStateVarInterface::new_from_children().into(),
-            boolean: SingleDependencyBooleanStateVarInterface::new(
-                BooleanState::get_value_dependency_instructions(),
-            )
-            .into(),
+            value: BooleanStateVar::new_from_children().into(),
```

Comment:
I'm a little worried how `BooleanStateVar` might be confused with `StateVar<bool>`. It's interesting how you can `BooleanStateVar.into()` to get a `StateVar<bool>`.

I can't decide if this similarity is a good thing so authors just do this naturally, or if it will cause confusion.

Reply:
The nice thing about Rust is that the types are inescapable, so the type information should always be with the developer. We could import them as `general_state_var::BooleanStateVar` if we really wanted to emphasize it.